### PR TITLE
Call ObjectStore::table_for_object_type() only when needed

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

`ObjectStore::table_for_object_type()` is an expensive call, and we should minimize the number of calls.

This closes https://jira.mongodb.org/browse/RJS-340

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: src/js_results.hpp [hunk 1]
```diff
@@ -160,17 +160,22 @@ inline void setup_aliases(parser::KeyPathMapping &mapping, const realm::SharedRe
 {
     const realm::Schema &schema = realm->schema();
     for (auto it = schema.begin(); it != schema.end(); ++it) {
+        TableRef table;
```
Comment (bmunkholm) [inline src/js_results.hpp]: Init to null.

Comment (kneth) [reply src/js_results.hpp]: I don't think it is needed since `TableRef`'s default constructor is setting it to `null`.

File: src/js_results.hpp [hunk 2]
```diff
@@ -160,17 +160,22 @@ inline void setup_aliases(parser::KeyPathMapping &mapping, const realm::SharedRe
...
         for (const Property &property : it->computed_properties) {
             if (property.type == realm::PropertyType::LinkingObjects) {
+                if (!table) {
```
Comment (cmelchior) [inline src/js_results.hpp]: Any reason this is still inside the loop of properties and not just defined where you define the `table` variable? It doesn't seem to change for each property?

Comment (kneth) [reply src/js_results.hpp]: `ObjectStore::table_for_object_type()` is an expensive call, and I only want to do it when needed (linking object property or a persisted property. The idea is to call it at most one time.

Comment (cmelchior) [reply src/js_results.hpp]: But we don't support schemas with no properties, right? So having the code twice just seems to reduce readability for no real benefit?

Comment (kneth) [reply src/js_results.hpp]: I see your point. Even in the case where a schema doesn't have any properties, we could do the call and avoid the branching. I mean, it is likely to be very rare that a schema doesn't have a property.

**Pull request merged**